pandas/tests/dtypes/test_inference.py

```diff
@@ -589,6 +590,24 @@ def test_maybe_convert_objects_bool_nan(self):
         out = lib.maybe_convert_objects(ind.values, safe=1)
         tm.assert_numpy_array_equal(out, exp)
 
+    def test_maybe_convert_objects_intenum(self):
+        class Colors(IntEnum):
+            red = 1
+            blue = 2
+
+        ind = pd.Index([Colors.red, Colors.blue], dtype=object)
+        exp = np.array([Colors.red, Colors.blue], dtype=object)
+        out = lib.maybe_convert_objects(ind.values)
+
+        # by default, we should not convert IntEnums to ints
+        tm.assert_numpy_array_equal(out, exp)
+
+        exp = np.array([1, 2], dtype=int)
+        out = lib.maybe_convert_objects(ind.values, convert_intenum=True)
+
+        # still coverts to int if convert_intenum set to True
+        tm.assert_numpy_array_equal(out, exp)
```

Comment:
For better readability and consistency with the majority of tests, I suggest to use ``result`` and ``expected`` variables rather than ``out`` and ``exp``.

Reply:
Ok, I will incorporate this improvement into my next commit.  I also need to fix an issue with the test on some Windows 32 bit machines.